integration_tests/src/examples.rs

```diff
@@ -299,3 +468,141 @@ fn get_formated_body(body_str: &str) -> String {
         msg_string,
     )
 }
+
+pub enum TestSample {
+    BigBuckBunny,
+    ElephantsDream,
+    Sample,
+    SampleLoop,
+    Generic,
+}
+
+pub enum TestCodec {
+    AAC,
+    LIBOPUS,
+}
+
+pub fn ff_stream_sample(
+    ip: &str,
+    video_port: Option<u16>,
+    audio_port: Option<u16>,
+    test_sample: TestSample,
+) -> Result<()> {
+    match test_sample {
+        TestSample::BigBuckBunny => ff_stream(
+            ip,
+            video_port,
+            audio_port,
+            get_asset_path(test_sample).unwrap(),
+        ),
+        TestSample::ElephantsDream => ff_stream(
+            ip,
+            video_port,
+            audio_port,
+            get_asset_path(test_sample).unwrap(),
+        ),
+        TestSample::Sample => {
+            if let Some(port) = video_port {
+                ff_stream_video(ip, port, get_asset_path(test_sample).unwrap())
+            } else {
+                Err(anyhow!("video port required for test sample"))
+            }
+        }
+        TestSample::SampleLoop => {
+            if let Some(port) = video_port {
+                ff_stream_video_loop(ip, port, get_asset_path(test_sample).unwrap())
+            } else {
+                Err(anyhow!("video port required for test sample"))
+            }
+        }
+        TestSample::Generic => {
+            if let Some(port) = video_port {
+                ff_stream_testsrc(
+                    ip,
+                    port,
+                    Resolution {
+                        width: 1920,
+                        height: 1080,
+                    },
+                )
+            } else {
+                Err(anyhow!("video port required for generic"))
+            }
+        }
+    }
+}
+
+// receives and plays h264 encoded video and opus encoded audio
```

Comment:
```suggestion
/// receives and plays h264 encoded video and opus encoded audio
```
triple slash is for docs, we don't generate docs, but those definitions still show up from LSP on however in editor